conda_build/skeletons/cran.py

```diff
@@ -594,17 +611,41 @@ def get_session(output_dir, verbose=True):
     return session
 
 
-def get_cran_metadata(cran_url, output_dir, verbose=True):
-    session = get_session(output_dir, verbose=verbose)
+def get_cran_archive_versions(cran_url, session, package, verbose=True):
+    if verbose:
+        print("Fetching archived versions for package %s from %s" % (package, cran_url))
+    r = session.get(cran_url + "/src/contrib/Archive/" + package + "/")
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        if e.response.status_code == 404:
+            print("No archive directory for package %s" % package)
+            return []
+        raise
+    versions = []
+    for p in re.findall(r'<td><a href="([^"]+)">\1</a></td>', r.text):
+        if p.endswith('.tar.gz') and '_' in p:
+            name, version = p.rsplit('.', 2)[0].split('_', 1)
+            versions.append(version)
+    return versions
+
+
+def get_cran_index(cran_url, session, verbose=True):
     if verbose:
-        print("Fetching metadata from %s" % cran_url)
-    r = session.get(cran_url + "/src/contrib/PACKAGES")
+        print("Fetching main index from %s" % cran_url)
+    r = session.get(cran_url + "/src/contrib/")
```

Comment:
This is a much lighter-weight download, and quicker to process. At the stages where we use this information we only need package name and version anyway.

Reply:
Ok, have you made measurements for this?

Reply:
Just my own subjective experience with this; it is building recipes is noticeably faster now. To be fair though I didn't A/B this one by itself. The speed difference in the use-binaries code path is significantly greater. There's a functional reason why this change is important, too—because I need to incorporate the _archived_ package into this index as well, and that can't be done by parsing a PACKAGES file.